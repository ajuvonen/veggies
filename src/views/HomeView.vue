<script setup lang="ts">
import {ref} from 'vue';
import {useRouter} from 'vue-router';
import {storeToRefs} from 'pinia';
import {DateTime} from 'luxon';
import {useActivityStore} from '@/stores/activityStore';
import ModalDialog from '@/components/ModalDialog.vue';
import BlueskyLink from '@/components/BlueskyLink.vue';

const router = useRouter();

const {startDate} = storeToRefs(useActivityStore());

const dialogOpen = ref(false);

const start = () => {
  startDate.value = DateTime.now().startOf('week');
  router.replace({name: 'log'});
};
</script>
<template>
  <div class="home__container">
    <h1 class="home__title">{{ $t('general.appTitle') }}</h1>
    <p class="sm:text-center">
      {{ $t('home.callout') }}
    </p>
    <div class="flex-container justify-center">
      <ButtonComponent data-test-id="home-info-button" @click="dialogOpen = true">{{
        $t('home.info')
      }}</ButtonComponent>
      <ButtonComponent data-test-id="home-start-button" @click="start()">{{
        $t('general.start')
      }}</ButtonComponent>
    </div>
    <BlueskyLink />
    <ModalDialog v-model="dialogOpen" :title="$t('home.infoTitle')">
      <template #content>
        <p>{{ $t('home.infoP1') }}</p>
        <p>{{ $t('home.infoP2') }}</p>
        <p>{{ $t('home.infoP3') }}</p>
      </template>
    </ModalDialog>
  </div>
</template>
<style scoped>
.home__container {
  @apply flex flex-col gap-6 items-center justify-center;
  @apply h-full;
}

.home__title {
  font-family: 'Bungee Shade', system-ui;
  font-size: 16vw;
  @apply sm:text-7xl sm:text-center uppercase;
}
</style>
