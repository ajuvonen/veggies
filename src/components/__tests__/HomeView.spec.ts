import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import HomeView from '@/views/HomeView.vue';
import DialogStub from './DialogStub.vue';

describe('HomeView', () => {
  it('renders', () => {
    const wrapper = mount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows dialog', async () => {
    const wrapper = mount(HomeView, {
      global: {
        stubs: {
          Dialog: DialogStub,
          DialogPanel: {
            template: '<div><slot /></div>',
          },
          DialogTitle: true,
        },
      },
    });
    expect(wrapper.findByTestId('dialog').exists()).toBe(false);
    await wrapper.findByTestId('home-info-button').trigger('click');
    expect(wrapper.findByTestId('dialog').exists()).toBe(true);
  });
});
