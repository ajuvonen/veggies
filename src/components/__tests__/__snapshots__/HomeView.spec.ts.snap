// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView > renders 1`] = `
"<div data-v-b4e148ca="" class="home__container">
  <h1 data-v-b4e148ca="" class="home__title">Eat Your Veggies</h1>
  <p data-v-b4e148ca="" class="sm:text-center">Adopt a healthy lifestyle by eating 30 plant-based ingredients each week. Track your progress easily with this free app.</p>
  <div data-v-b4e148ca="" class="flex-container justify-center"><button data-v-24e38f3d="" data-v-b4e148ca="" class="button button--primary" data-test-id="home-info-button">
      <!--v-if-->Learn more
    </button><button data-v-24e38f3d="" data-v-b4e148ca="" class="button button--primary" data-test-id="home-start-button">
      <!--v-if-->Start
    </button></div><a data-v-ba48d2a1="" data-v-b4e148ca="" class="bluesky-link" href="https://bsky.app/profile/eatyourveggies.app" target="_blank" noreferrer="" noopener=""><svg data-v-ba48d2a1="" style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M13 21H11V6L9.03 3.97L10 3L12 5L14 3L15 4L13 6V21M14 9H14V7C14.84 6.37 15.88 6 17 6C19.76 6 22 8.24 22 11C22 12.64 21.21 14.09 20 15H20V17C20 19.21 18.21 21 16 21C15.27 21 14.59 20.81 14 20.46V18.46C14.59 18.81 15.27 19 16 19C17.1 19 18 18.1 18 17V13.83C19.17 13.42 20 12.31 20 11C20 9.35 18.67 8 17 8H17C15.88 8 14.84 8.37 14 9M10 9H10C9.16 8.37 8.13 8 7 8H7C5.33 8 4 9.35 4 11C4 12.31 4.84 13.42 6 13.83V17C6 18.1 6.9 19 8 19C8.73 19 9.41 18.81 10 18.46V20.46C9.41 20.81 8.73 21 8 21C5.79 21 4 19.21 4 17V15H4C2.79 14.09 2 12.64 2 11C2 8.24 4.24 6 7 6C8.13 6 9.16 6.37 10 7V9M9 11L7.5 9.5L6 11L7.5 12.5L9 11M18 11L16.5 9.5L15 11L16.5 12.5L18 11Z" class="transition-all"></path>
    </svg> Follow on Bluesky</a>
  <!---->
</div>"
`;
